ipv8/community.py

```diff
@@ -180,40 +139,23 @@ def on_deprecated_message(self, source_address, data):
                             self.deprecated_message_names[data[22]], *source_address)
 
     def bootstrap(self):
-        if time() - self.last_bootstrap < BOOTSTRAP_TIMEOUT:
-            return
-        self.logger.debug("Bootstrapping %s, current peers %d", self.__class__.__name__, len(self.get_peers()))
-        self.last_bootstrap = time()
-        for socket_address in _DEFAULT_ADDRESSES:
-            self._ensure_blacklisted(socket_address)
-            self.walk_to(socket_address)
-
-    def _ensure_blacklisted(self, address):
+        if not self._bootstrappers_initialized:
+            self.bootstrappers = [bootstrapper for bootstrapper in self.bootstrappers if bootstrapper.initialize(self)]
```

Comment:
It may be useful to have `initialize` async as well. If this can be done cleanly.

Reply:
Implemented 👍